test/e2e/systemd_activate_test.go

```diff
@@ -103,4 +105,37 @@ var _ = Describe("Systemd activate", func() {
 		Expect(abiSession).To(Exit(0))
 		Expect(abiSession.OutputToString()).To(Equal("true"))
 	})
+
+	It("invalid systemd file descriptor", func() {
+		host := "127.0.0.1"
+		port, err := podmanUtils.GetRandomPort()
+		Expect(err).ToNot(HaveOccurred())
+
+		addr := fmt.Sprintf("%s:%d", host, port)
+
+		// start systemd activation with datagram socket
+		activateSession := testUtils.StartSystemExec(activate, []string{
+			"--datagram", "--listen", addr,
+			podmanTest.PodmanBinary,
+			"--root=" + filepath.Join(tempDir, "server_root"),
+			"system", "service",
+			"--time=0",
+		})
+		Expect(activateSession.Exited).ShouldNot(Receive(), "Failed to start podman service")
+
+		// we have to wait for systemd-socket-activate to become ready
+		time.Sleep(1 * time.Second)
```

Comment:
Can we do an exponential backoff here?

Reply:
Well the problem is there is no real way to know when it is ready. Because it uses udp it succeeds always so I had to add a sleep to make it work.